lib/numerizer.rb

```diff
@@ -108,50 +105,105 @@ class Numerizer
         string.gsub!(/(^|\W)#{tp[0]}#{dn[0]}(?=$|\W)/i, '\1<num>' + (tp[1] + dn[1]).to_s)
       end
-      SINGLE_ORDINALS.each do |dn|
+      (ORDINALS + SINGLE_ORDINAL_FRACTIONALS).each do |dn|
+        next if ignore.include? dn[0].downcase 
         string.gsub!(/(^|\W)#{tp[0]}(\s)?#{dn[0]}(?=$|\W)/i, '\1<num>' + (tp[1] + dn[1]).to_s + dn[0][-2, 2])
       end
       string.gsub!(/(^|\W)#{tp[0]}(?=$|\W)/i, '\1<num>' + tp[1].to_s)
     end
 
     # handle fractions
-    FRACTIONS.each do |tp|
+    # only plural fractions if ordinal mode
+    if bias == :ordinal
+      fractionals = ONLY_PLURAL_FRACTIONS 
+    else
+      fractionals = ALL_FRACTIONS
+    end
+
+    fractionals.each do |tp|
+      next unless ignore.select {|x| /#{tp[0]}/ =~ x } .empty?
       string.gsub!(/a #{tp[0]}(?=$|\W)/i, '<num>1/' + tp[1].to_s)
-      string.gsub!(/\s#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      # TODO : Find Noun Distinction for Quarter
+      # Handle Edge Case with Quarter
+      if bias == :fractional
+        string.gsub!(/(^|\W)#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      elsif /quarter/ =~ tp[0] then
+        string.gsub!(/(?:\w*)(^|\W)#{tp[0]}(?=$|\W)/i) do |match|
+          if (match =~ /^(i|you|he|she|we|it|you|they|to|the)/i) == nil then 
+            match.gsub!(/(^|\W)#{tp[0]}/, '/' + tp[1].to_s)
+          end
+          match
+        end
+      else
+        string.gsub!(/(?<!the)\s#{tp[0]}(?=$|\W)/i, '/' + tp[1].to_s)
+      end
     end
 
-    (DIRECT_ORDINALS + SINGLE_ORDINALS).each do |on|
-      string.gsub!(/(^|\W)#{on[0]}(?=$|\W)/i, '\1<num>' + on[1].to_s + on[0][-2, 2])
+    ALL_ORDINALS.each do |on|
+      break if bias == :fractional # shouldn't be necessary but saves cycles
+      next if ignore.include? on[0].downcase 
+      if on[0] == 'second' and bias != :ordinal then
+        # We don't want to substitute second preceeded by second or another ordinal
+        string.gsub!(/\w*(?<!second)(^|\W)#{on[0]}(?=$|\W)/i) do |match| 
+          unless match =~ /^(\d|#{ALL_ORDINALS_REGEX})/ then
+            match.gsub!(/(^|\W)#{on[0]}(?=$|\W)/i, '\1<num>' + on[1].to_s + on[0][-2, 2])
+          end
+          match
+        end
+      else
+        string.gsub!(/(^|\W)#{on[0]}(?=$|\W)/i, '\1<num>' + on[1].to_s + on[0][-2, 2])
+      end
     end
 
     # evaluate fractions when preceded by another number
     string.gsub!(/(\d+)(?: | and |-)+(<num>|\s)*(\d+)\s*\/\s*(\d+)/i) { ($1.to_f + ($3.to_f/$4.to_f)).to_s }
 
+    # fix unpreceeded fractions
+    string.gsub!(/(?:^|\W)\/(\d+)/, '1/\1')
+    string.gsub!(/(?<=[a-zA-Z])\/(\d+)/, ' 1/\1')
+
     # hundreds, thousands, millions, etc.
     BIG_PREFIXES.each do |bp|
+      next if ignore.include? bp[0].downcase 
       string.gsub!(/(?:<num>)?(\d*) *#{bp[0]}/i) { $1.empty? ? bp[1] : '<num>' + (bp[1] * $1.to_i).to_s }
       andition(string)
     end
 
     andition(string)
 
+    # always substitute halfs
+    unless ignore.include? 'half' then
```

Comment:
You don't need the `then`, you have this in a few places and it can be dropped in all of them.